stop_and_wait_plugin/src/stop_and_wait_plugin.cpp

```diff
@@ -217,7 +226,18 @@ namespace stop_and_wait_plugin
                         throw std::invalid_argument("Ending distance is beyond known route"); 
                     }
                 }
-                else jerk_ = jerk_req;
+                else {
+                    //find distance to the end
+                    lanelet::BasicPoint2d curr_pose (pose_msg_.pose.position.x,pose_msg_.pose.position.y);
+                    double current_downtrack = wm_->routeTrackPos(curr_pose).downtrack;
+                    //stay approximately at crawl speed until within destination downtrack range (defined in route)
+                    if(start_speed <= min_crawl_speed_ && current_downtrack < ending_downtrack - 10.0){
```

Comment:
10.0 should probably be a parameter or loaded directly from the route end distance parameter?

Reply:
Yes, this can be loaded from the param in route